### PR TITLE
[Do Not Merge] [Experiment] Excise SwiftOnoneSupport

### PR DESCRIPTION
**Pull request by harlanhaskins**
Just curious to see how much of an impact it has, and if maybe we can replace it instead with `@_specialize` attributes in the stdlib.

### CODE REVIEW

**Comment by jrose-apple** [comment]
Smoke benchmark only runs -O tests, so it won't give you the info you need.

@swift-ci Please benchmark

**Comment by harlanhaskins** [comment]
@swift-ci Please benchmark

File: test/Driver/emit-sil-single-file.swift [hunk 1]
```diff
@@ -12,14 +12,6 @@
 // CHECK: Hello World
 // CHECK: Hello Bob, today is Tuesday.
 
-// This test intentionally mirrors /Driver/emit-sib-single-file to ensure that
-// SwiftOnoneSupport is always a dependency of -Onone -emit-si*gen builds.
```
Comment (CodaFi) [inline test/Driver/emit-sil-single-file.swift]: You can either collapse this and its SIBGen sibling together and move them with other executable tests or straight-up remove them after you’re done here.

Comment (harlanhaskins) [reply test/Driver/emit-sil-single-file.swift]: Yeah I was hoping to just remove them in The Real Patch™

**Comment by harlanhaskins** [comment]
@swift-ci please benchmark

**Comment by CodaFi** [comment]
@swift-ci please test compiler performance

**Comment by harlanhaskins** [comment]
Absolutely incredible. Now it’s a matter of figuring out how much performance we can gain back through explicitly specializing.

**Pull request closed without merging**